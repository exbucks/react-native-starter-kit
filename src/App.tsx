import React from 'react';
import { Provider } from 'react-redux'
import { PersistGate } from 'redux-persist/integration/react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import configureStore from './store';

import HomeScreen from './containers/home'
import LoginScreen from './containers/login'
import RegisterScreen from './containers/register'

type RootStackParamList = {
  Login: undefined;
  Register: undefined;
  Home: {userId: string, sort: 'latest' | 'top'};
}

const RootStack = createNativeStackNavigator<RootStackParamList>();
const { store, persistor } = configureStore()

function App(): JSX.Element {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <NavigationContainer>
          <RootStack.Navigator initialRouteName="Login">
            <RootStack.Screen name="Home" component={HomeScreen} initialParams={{ userId: '0', sort: 'latest' }} />
            <RootStack.Screen name="Login" component={LoginScreen} />
            <RootStack.Screen name="Register" component={RegisterScreen} />
          </RootStack.Navigator>
        </NavigationContainer>
      </PersistGate>
    </Provider>
  );
}

export default App;
