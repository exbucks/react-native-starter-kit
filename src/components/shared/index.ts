export * from './text'
export * from './keyboard'
export * from './drawer'
export * from './header'
export * from './progress'
export * from './icon'
