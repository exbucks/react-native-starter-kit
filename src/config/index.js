import { Text } from 'react-native'
import AppConfig from './app'

// Allow/disallow font-scaling in app
Text.defaultProps.allowFontScaling = AppConfig.allowTextFontScaling

if (__DEV__) {
  // If ReactNative's yellow box warnings are too much, it is possible to turn
  // it off, but the healthier approach is to fix the warnings.  =)
}
