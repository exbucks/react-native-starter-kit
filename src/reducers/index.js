import { createStore, combineReducers } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import storage from 'redux-persist/lib/storage'
import configureStore from './createStore'
import rootSaga from '../sagas'
import immutablePersistenceTransform from '../services/transform'

const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['nav'],
  transforms: [immutablePersistenceTransform],
}

/* ------------- Assemble The Reducers ------------- */
const reducers = combineReducers({
  nav: require('./navigation').reducer,
  app: require('./app').reducer,
})

const persistedReducer = persistReducer(persistConfig, reducers)

let { store, sagasManager, sagaMiddleware } = configureStore(persistedReducer, rootSaga)

if (module.hot) {
  module.hot.accept(() => {
    const nextRootReducer = require('./').reducers
    store.replaceReducer(nextRootReducer)

    const newYieldedSagas = require('../sagas').default
    sagasManager.cancel()
    sagasManager.done.then(() => {
      sagasManager = sagaMiddleware.run(newYieldedSagas)
    })
  })
}

let persistor = persistStore(store)

export { store, persistor }
