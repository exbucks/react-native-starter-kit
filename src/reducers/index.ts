import {combineReducers} from 'redux'
import appReducer from './app'
import isLoadingReducer from './isLoading'
import errorReducer from './error'

const rootReducer = combineReducers({
    app: appReducer,
    isLoading: isLoadingReducer,
    error: errorReducer
})

export type AppState = ReturnType<typeof rootReducer>

export default rootReducer
