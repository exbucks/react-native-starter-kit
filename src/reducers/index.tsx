import { combineReducers } from 'redux'
import app from './app'
import dev from './dev'

export default combineReducers({
    app,
    dev
})