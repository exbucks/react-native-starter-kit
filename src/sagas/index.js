import { takeLatest, all } from 'redux-saga/effects'

/**
 * Types of actions
 */
import { AppTypes } from '../actions/app'

/**
 * Sagas
 */
import { loginRequest } from './app'

/**
 * API
 */
// The API we use is only used from Sagas, so we create it here and pass along
// to the sagas which need it.
const api = () => {}

/**
 * Connect Types to Sagas
 */
export default function* root() {
  yield all([
    // some sagas receive extra parameters in addition to an action
    takeLatest(AppTypes.LOGIN_REQUEST, loginRequest, api),
  ])
}
