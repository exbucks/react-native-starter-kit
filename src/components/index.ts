import { DrawerScreen, Text, KeyboardSpacer, Header, ProgressBar, Icon } from './shared'

export { DrawerScreen, Text, KeyboardSpacer, Header, ProgressBar, Icon }
