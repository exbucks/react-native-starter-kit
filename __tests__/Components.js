import 'react-native'
import React from 'react'
import { Text } from '../src/components'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'


it('Text renders correctly', () => {
  const tree = renderer.create(
    <Text />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
