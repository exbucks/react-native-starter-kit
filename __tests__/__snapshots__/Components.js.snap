// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#000000",
      "fontSize": 16,
    }
  }
/>
`;
